tasks/vmaas_sync/package_sync.go

```diff
@@ -218,15 +235,62 @@ func storePackageDetailsFrmPkgListItems(tx *gorm.DB, nameIDs map[string]int, pkg
 	}
 	utils.Log("packages", len(toStore)).Info("Collected packages to store")
 
+	err := toStore.storeOrUpdate(tx)
+	return err
+}
+
+//nolint:funlen
+func (pkgs pkgSlice) storeOrUpdate(tx *gorm.DB) error {
+	var toUpdate pkgSlice
+
+	nameIDEvra := make([][]interface{}, 0, len(pkgs))
```

Comment:
Maybe this would be better?
```suggestion
	nameIDEVRAs := make([]nameIDandEvra, 0, len(pkgs))
```

Reply:
I would refactor this old code `nameIDandEvra`, but it's up to you

Reply:
will it work for gorm? 
```
tx.Where("(name_id, evra) IN ?", nameIDEvra).Find(&toUpdate)
```

Reply:
doesn't seem to work with gorm